test/type-tests/bind-all.usage-types.ts

```diff
@@ -23,13 +23,13 @@ import { bindAll } from '../../src';
   bindAll(button, [
     {
       type: 'click',
-      listener(event: MouseEvent) {
+      listener(event) {
```

Comment:
Those now should be correctly inferred. Note that with this PR we lose the autocomplete in 4.x but I think that is a fair tradeoff. It will just start working in 5.0. 

Please test it with TS 5.0 (beta) before merging.